riptide-failsafe/src/main/java/org/zalando/riptide/failsafe/CompositeDelayFunction.java

```diff
@@ -14,23 +14,28 @@
 
 @API(status = EXPERIMENTAL)
 @AllArgsConstructor
-public final class CompositeDelayFunction<R, X extends Throwable>
-        implements DelayFunction<R, X> {
+public final class CompositeDelayFunction<R> implements ContextualSupplier<R, Duration> {
 
-    private final Collection<DelayFunction<R, X>> functions;
+    private final Collection<ContextualSupplier<R, Duration>> functions;
 
     @Override
-    public Duration computeDelay(final R result, final X failure, final ExecutionContext context) {
+    public Duration get(final ExecutionContext<R> context) throws Throwable {
         return functions.stream()
-                .map(function -> function.computeDelay(result, failure, context))
+                .map(function -> {
+                    try {
```

Comment:
Given that the method signature changed to throw instances of `Throwable`, why do we need to wrap this into a `RuntimeException`?

Reply:
`map` method can't accept function that throws checked exception